lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -386,6 +389,24 @@ function (array $fieldMapping) use ($class): ?string {
                             return null;
                         }
 
+                        if (
+                            is_a($propertyType, BackedEnum::class, true)
+                            && $metadataFieldType === get_debug_type($propertyType::cases()[0]->value)
```

Comment:
`get_debug_type` has a name that indicates it is probably not intended for this. Is there a way to do the same with the reflection API?

Reply:
Also, why does the backing type matter? As soon as the property type enum is not the exact same as the mapping enum, it's an issue, isn't it?

Reply:
> get_debug_type has a name that indicates it is probably not intended for this. Is there a way to do the same with the reflection API?

Sure, using `ReflectionEnum` instead

Reply:
> Also, why does the backing type matter? As soon as the property type enum is not the exact same as the mapping enum, it's an issue, isn't it?

Still, we need to check that the backing type matches the mapping type. The checking order is as it's due to different validation message. I've added a test case to check that.